test/integration/scheduler/util.go

```diff
@@ -0,0 +1,334 @@
+	nodes := make([]*v1.Node, numNodes)
+	for i := 0; i < numNodes; i++ {
+		nodeName := fmt.Sprintf("%v-%d", prefix, i)
+		node, err := createNode(cs, nodeName)
+		if err != nil {
+			return nodes[:], err
+		}
+		nodes[i] = node
+	}
+	return nodes[:], nil
+}
+
+type pausePodConfig struct {
+	Name                              string
+	Namespace                         string
+	Affinity                          *v1.Affinity
+	Annotations, Labels, NodeSelector map[string]string
+	Resources                         *v1.ResourceRequirements
+	Tolerations                       []v1.Toleration
+	NodeName                          string
+	SchedulerName                     string
+}
+
+// initPausePod initializes a pod API object from the given config. It is used
+// mainly in pod creation process.
+func initPausePod(cs clientset.Interface, conf *pausePodConfig) *v1.Pod {
+	pod := &v1.Pod{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:        conf.Name,
+			Labels:      conf.Labels,
+			Annotations: conf.Annotations,
+		},
+		Spec: v1.PodSpec{
+			NodeSelector: conf.NodeSelector,
+			Affinity:     conf.Affinity,
+			Containers: []v1.Container{
+				{
+					Name:  conf.Name,
+					Image: framework.GetPauseImageName(cs),
+				},
+			},
+			Tolerations:   conf.Tolerations,
+			NodeName:      conf.NodeName,
+			SchedulerName: conf.SchedulerName,
+		},
+	}
+	if conf.Resources != nil {
+		pod.Spec.Containers[0].Resources = *conf.Resources
+	}
+	return pod
+}
+
+// createPausePod creates a pod with "Pause" image and the given config and
+// return its pointer and error status.
+func createPausePod(cs clientset.Interface, conf *pausePodConfig) (*v1.Pod, error) {
+	p := initPausePod(cs, conf)
+	return cs.CoreV1().Pods(conf.Namespace).Create(p)
+}
+
+// createPausePodWithResource creates a pod with "Pause" image and the given
+// resources and returns its pointer and error status.
+func createPausePodWithResource(cs clientset.Interface, podName string, nsName string, res *v1.ResourceList) (*v1.Pod, error) {
+	conf := pausePodConfig{
+		Name:      podName,
+		Namespace: nsName,
+		Resources: &v1.ResourceRequirements{
+			Requests: *res,
+		},
+	}
+	return createPausePod(cs, &conf)
+}
+
+// createDefaultPausePod creates a pod with "Pause" image and returns its pointer
+// and error status.
+func createDefaultPausePod(cs clientset.Interface, podName string, nsName string) (*v1.Pod, error) {
+	conf := pausePodConfig{
+		Name:      podName,
+		Namespace: nsName,
+	}
+	return createPausePod(cs, &conf)
+}
+
+// runPausePod creates a pod with "Pause" image and the given config and waits
+// until it is scheduled. It returns its pointer and error status.
+func runPausePod(cs clientset.Interface, conf *pausePodConfig) (*v1.Pod, error) {
+	p := initPausePod(cs, conf)
+	pod, err := cs.CoreV1().Pods(conf.Namespace).Create(p)
+	if err != nil {
+		return nil, fmt.Errorf("Error creating pause pod: %v", err)
+	}
+	if err = waitForPodToSchedule(cs, pod); err != nil {
```

Comment:
It wouldn't be more or less the same. err is already defined in the code. So, I wouldn't worry about it.